flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -118,40 +120,38 @@ public MetricRegistry(Configuration config) {
 
 				if (reporter instanceof Scheduled) {
 					LOG.info("Periodically reporting metrics in intervals of {} {}", period, timeunit.name());
-					long millis = timeunit.toMillis(period);
 					
-					timer = new java.util.Timer("Periodic Metrics Reporter", true);
-					timer.schedule(new ReporterTask((Scheduled) reporter), millis, millis);
+					executor.scheduleWithFixedDelay(new ReporterTask((Scheduled) reporter), period, period, timeunit);
 				}
 				else {
-					timer = null;
+					executor = null;
 				}
 			}
 			catch (Throwable t) {
 				reporter = new JMXReporter();
-				timer = null;
+				executor = null;
 				LOG.error("Could not instantiate custom metrics reporter. Defaulting to JMX metrics export.", t);
 			}
 
 			this.reporter = reporter;
-			this.timer = timer;
+			this.executor = executor;
 		}
 	}
 
 	/**
 	 * Shuts down this registry and the associated {@link org.apache.flink.metrics.reporter.MetricReporter}.
 	 */
 	public void shutdown() {
-		if (timer != null) {
-			timer.cancel();
-		}
 		if (reporter != null) {
 			try {
 				reporter.close();
 			} catch (Throwable t) {
 				LOG.warn("Metrics reporter did not shut down cleanly", t);
 			}
 		}
+		if (executor != null) {
+			executor.shutdownNow();
```

Comment:
I guess that one second of delay wouldn't really hurt. That's at least how the `ScheduledReporter` do it.

Btw: Why do we actually replicate the execution logic for the `com.codahale.metrics.ScheduledReporter`. As far as I can tell, we could simply use the internal reporting mechanism. Then we would not have to implement the `Scheduled` interface.